ml-agents/mlagents/trainers/stats.py

```diff
@@ -213,9 +213,12 @@ def add_property(
             assert isinstance(value, dict)
             text = self._dict_to_tensorboard("Hyperparameters", value)
             self._maybe_create_summary_writer(category)
-            self.summary_writers[category].add_summary(text, 0)
+            if text is not None:
+                self.summary_writers[category].add_summary(text, 0)
 
-    def _dict_to_tensorboard(self, name: str, input_dict: Dict[str, Any]) -> str:
+    def _dict_to_tensorboard(
+        self, name: str, input_dict: Dict[str, Any]
+    ) -> Optional[str]:
```

Comment:
I don't think this actually returns a string; FileWriter.add_summary takes a Summary proto obj: https://www.tensorflow.org/api_docs/python/tf/compat/v1/summary/FileWriter#add_summary

Reply:
It says "optionally serialized as a string. I am not sure what tf.summary.text operator is returning after sess.run but it could be a protobuf

Reply:
I printed the type and it looks like it is `<class 'bytes'>`

Reply:
Yup, was just about to say that.